webapp/packages/plugin-data-import/src/DataImportDialog/useDataImportDialog.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { action, observable } from 'mobx';
+
+import { useObservableRef } from '@cloudbeaver/core-blocks';
+import type { DataTransferProcessorInfo } from '@cloudbeaver/core-sdk';
+
+import { EDataImportDialogStep } from './EDataImportDialogStep';
+import type { IDataImportDialogState } from './IDataImportDialogState';
+
+interface IDialog {
+  state: IDataImportDialogState;
+  stepBack: () => void;
+  selectProcessor: (processor: DataTransferProcessorInfo) => void;
+  deleteFile: (id: string) => void;
+  reset: () => void;
+}
+
+const DEFAULT_STATE_GETTER: () => IDataImportDialogState = () => ({
+  step: EDataImportDialogStep.PROCESSOR,
+  files: null,
+  selectedProcessor: null,
+});
+
+export function useDataImportDialog(initialState?: IDataImportDialogState) {
+  const dialog = useObservableRef<IDialog>(
+    () => ({
+      state: initialState ?? DEFAULT_STATE_GETTER(),
+      stepBack() {
```

Comment:
this one should be renamed to something like goFirstStep

or we can define here a system of steps to easy implement few more steps when we will need it

I mean
```
const steps = [
    {
        id: EDataImportDialogStep.PROCESSOR, 
        component: <ProcessorComponent/>
    },
    {
        id: EDataImportDialogStep.FILE, 
        component: <FileComponent/>
    }, 
    ...
]
```

and then in this method you can find which step we are currently on, an go to real prev step

Reply:
or use switch (not sure that we need steps list abstraction here

Reply:
for now yes, it is unneeded changes. so better just to rename